src/core/dbus-execute.c

```diff
@@ -2225,7 +2225,7 @@ int bus_exec_context_set_transient_property(
                         if (!credential_glob_valid(glob))
                                 return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Credential name or glob is invalid: %s", glob);
 
-                        if (rename && !credential_name_valid(rename))
+                        if (!isempty(rename) && !credential_name_valid(rename))
```

Comment:
Please use empty_to_null() to normalize rename here. Otherwise, the transient unit written to disk would contain extraneous colon.

Reply:
Ah, indeed. Thanks.